lib/model: Support removing paused folders (fixes #4357)

AudriusButkevicius opened the pull request:
Tested manually.

imsodin commented on lib/model/model.go:
```diff
@@ -334,7 +334,15 @@ func (m *Model) RemoveFolder(folder string) {
+	}
+	if !ok {
+		// Folder does not exist... Weird...
+		return
```
I think this should never happen and thus panic

AudriusButkevicius replied:
When/if we add `HTTP DELETE /folders/xyz` API, this would make sense.

imsodin replied:
Shouldn't that result in a returned error? I think there are some endpoints presently that check whether a folder is present before doing anything.

imsodin commented:
Test:

```go
func TestRemovePausedFolder(t *testing.T) {
	// issue 4357

	db := db.OpenMemory()
	m := NewModel(defaultConfig, protocol.LocalDeviceID, "syncthing", "dev", db, nil)
	defCfg := defaultFolderConfig.Copy()
	m.AddFolder(defCfg)
	m.StartFolder("default")
	m.ServeBackground()
	defer m.Stop()

	oldCfg := m.cfg.RawCopy()
	newCfg := oldCfg.Copy()
	newCfg.Folders[0].Paused = true
	m.CommitConfiguration(oldCfg, newCfg)

	m.RemoveFolder("default")
	m.fmut.Lock()
	if _, ok := m.folderCfgs["default"]; ok {
		t.Error("Paused folder was not removed")
	}
	m.fmut.Unlock()
}
```

AudriusButkevicius commented:
I think the test is not very useful. CommitConfig does the removal from folderCfgs, not the m.RemoveFolder in this case.

AudriusButkevicius commented:
Yet let me add a different one, and add the panic I guess.

imsodin commented:
> I think the test is not very useful. CommitConfig does the removal from folderCfgs, not the m.RemoveFolder in this case.

It panics as in the referenced issue, so I guess it does what it is supposed to: reproduce it - what am I missing?

AudriusButkevicius commented:
My point was that this check was too late:
```
if _, ok := m.folderCfgs["default"]; ok {
	t.Error("Paused folder was not removed")
}
```

Anyways, updated and added a test case.

imsodin commented:
@st-release lgtm

imsodin commented:
@st-review lgtm

st-review commented:
@imsodin: Noted! Need another LGTM or explicit merge command.

calmh commented:
@st-review lgtm

st-review commented:
:ok_hand: Merged as e85ce7c94e6e6ee537546c5c88a93d6f997a9d3b. Thanks, @AudriusButkevicius!
